app/src/main/java/com/android/voyageur/model/notifications/TripInviteRepositoryFirebase.kt

```diff
@@ -0,0 +1,196 @@
+package com.android.voyageur.model.notifications
+
+import com.google.firebase.firestore.AggregateSource
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.ListenerRegistration
+import com.google.firebase.firestore.SetOptions
+
+/**
+ * Used for displaying trip invites and notifications Appears as a parameter to the ${UserViewModel}
+ *
+ * @param db the firestore instance to use
+ * @param collectionPath the path to the collection of trip invites
+ * @constructor creates a new TripInviteRepositoryFirebase
+ * @property db the firestore instance to use
+ * @property collectionPath the path to the collection of trip invites
+ */
+class TripInviteRepositoryFirebase(private val db: FirebaseFirestore) : TripInviteRepository {
+  private val collectionPath = "friendRequests"
+
+  override fun init(onSuccess: () -> Unit) {
+    db.collection(collectionPath).get().addOnSuccessListener { onSuccess() }
+  }
+
+  /**
+   * Get the number of trip invites for a user
+   *
+   * @param userId the user to get the number of trip invites for
+   * @param onSuccess callback to execute after successful fetch
+   * @param onFailure exception handling callback
+   */
+  override fun getTripInvitesCount(
+      userId: String,
+      onSuccess: (Long) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection(collectionPath)
+        .whereEqualTo("to", userId)
+        .count()
+        .get(AggregateSource.SERVER)
+        .addOnSuccessListener { onSuccess(it.count) }
+        .addOnFailureListener { exception -> onFailure(exception) }
+  }
+
+  /**
+   * Create a trip invite
+   *
+   * @param req the trip invite to create
+   * @param onSuccess callback to execute after successful creation
+   * @param onFailure exception handling callback
+   */
+  override fun createTripInvite(
+      req: TripInvite,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection(collectionPath)
+        .document(req.id)
+        .set(req, SetOptions.merge())
+        .addOnSuccessListener { onSuccess() }
+        .addOnFailureListener { exception -> onFailure(exception) }
+  }
```

Comment:
You are assuming that req is always valid, but it would be safer to include a null or empty check to handle potential issues. We’ve encountered bugs in the past caused by invalid input, which could arise from network errors, data corruption, or deserialization issues. Additionally, malformed or incomplete data from external sources(e.g., third-party integrations or API calls) could lead to unexpected behavior. Adding a validation step here would make the coee more robust and resilient to these scenarios, reducing the likelihood of similar issues in the future.

Reply:
Yes, find the whole conversation [here](https://github.com/swent-t16/Voyageur/pull/290/files/d8855fb0be99a8bcf960272e1be63df275b28a64#r1874999422)